modules/_node-scripts/util/filterChangedFiles.mjs

```diff
@@ -0,0 +1,57 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {$} from 'execa';
+import path from 'path';
+
+/**
+ * In the context of liferay-portal, we may want to run against a subset of
+ * eligible files (eg. files changed on the current branch). To achieve this, we
+ * check the LIFERAY_NPM_SCRIPTS_WORKING_BRANCH_NAME environment variable and if
+ * it is set, filter the `files` list to contain only files changed with respect
+ * to that branch (usually "master", but may also be "master-private").
+ *
+ * If the variable is not set, the `files` list is return unchanged.
+ *
+ * One important exception to the above: if the top-level `package.json`
+ * changes (which happens rarely), this may indicate a change of the
+ * @liferay/npm-scripts version, and in that case we want to run against
+ * the entire unfiltered `files` list.
+ *
+ * @param {Array<string>} files List of files relative to the current directory.
+ */
+export default async function filterChangedFiles(files) {
```

Comment:
This one seems generic enough to be put in the global `util` directory (since it can be used from other tasks).

One "pattern" I was applying in other PRs is to put `util` subdirectories inside tasks' folders. I put util functions that don't seem generic enough to be put in the global `util` but that may one day be promoted there for some reason.

If we follow that pattern it may be easier for us to search in the util directories of tasks before duplicating helper functions (maybe). What do you think?